src/loader/timescale_bgw.c

```diff
@@ -0,0 +1,661 @@
+#include <postgres.h>
+
```

Comment:
I see three components here. 
1) The "bgw" module which connects all bgw stuff this can stay in "timescale_bgw.c" (actually "bgw.c" is better since we don't prefix other files with "timescale"). This includes the shmem_init/alloc hooks/ the sql functions etc.
2) The worker counter and associated shared state (worker_counter.c/h?)
3) The launcher worker stuff (bgw_launcher.c/h?)

Reply:
So, I thought we'd decided that we wanted to have a distinctive prefix due to your concerns about polluting the global namespace. Perhaps we can just go with tsbgw.c. I do feel that bgw.c is a bit more likely to have some sort of collision.